src/RadixDlt.NetworkGateway.PostgresIntegration/LedgerExtension/PostgresLedgerExtenderService.cs

```diff
@@ -1148,6 +1148,32 @@ nonFungibleResourceManagerFieldIdTypeSubstate.Value.NonFungibleIdType switch
 
                             vaultSnapshots.Add(new NonFungibleVaultSnapshot(referencedEntity, resourceEntity, simpleRep, true, stateVersion));
                         }
+
+                        if (substateId.SubstateType == CoreModel.SubstateType.PackageCodeVmTypeEntry)
+                        {
+                            var keyHex = (substateId.SubstateKey as CoreModel.MapSubstateKey)!.KeyHex;
```

Comment:
I guess this cast is mandatory? `((CoreMode.MapSubstateKey)substateKey).KeyHex` would be cleaner?